### PR TITLE
provider/maas: only bridge specified interfaces

### PR DESCRIPTION
**Pull request by frobware**
The list of interfaces to be bridged is no longer automatically
determined by the bridge script; you know need to specify a space
separated list of interface names that you want bridging.

For example:

 $ add-juju-bridge.py \
     --interfaces-to-bridge="eth0 eth0:1 eth0:2 bond2" /e/n/i

The MAAS provider now works out which interfaces are to be bridged and
invokes the script with a specific set of interface names. The MAAS
provider will now bridge any interface that has a subnet and is not
unconfigured (i.e., it has an IP address).

Fixes [LP:#1627037](https://bugs.launchpad.net/juju/+bug/1627037)

### CODE REVIEW

File: provider/maas/environ.go [hunk 1]
```diff
@@ -1052,6 +1062,55 @@ func (environ *maasEnviron) StartInstance(args environs.StartInstanceParams) (
 	}, nil
 }
 
+func instanceConfiguredInterfaceNames(usingMAAS2 bool, inst instance.Instance, subnetsMap map[string]network.Id) ([]string, error) {
+	var (
+		interfaces []network.InterfaceInfo
+		err        error
+	)
+	if !usingMAAS2 {
```
Comment (dooferlad) [inline provider/maas/environ.go]: Pass in MAAS API version and switch.

**Comment by frobware** [comment]
We're going to drop this PR and produce a final-final version.

File: provider/maas/environ.go (hunk 1 above)
Comment (frobware) [inline provider/maas/environ.go]: We should do this as a separate change; there are many other places that pass in / use the result of usingMAAS2().

**Comment by frobware** [comment]
And... having checked everything we're now going to keep this PR.

**Comment by frobware** [comment]
I have tested this MAAS 2.0 using a combination of nodes with:

- one nic
- two nics, one unconfigured
- bonded nics, parent unconfigued
- vlan on a bond

And for each scenario I added a LXD container. In one setup I ran into the issue PR #6276 fixes, and indeed it fixed my setup too. For each of these scenarios I visually inspected /etc/network/interfaces on both the host and in the container.

Tested the combinations using xenial.

**Comment by frobware** [comment]
I have also tested with trusty.

**Comment by dimitern** [comment]
I tested with trusty on 1.9 MAAS as well, and with xenial (yesterday) on both 2.0 and 1.9 MAAS

File: provider/maas/environ.go (hunk 1 above)
Comment (dooferlad) [inline provider/maas/environ.go]: OK, :+1: in that case.

**Comment by mitechie** [comment]
$$merge$$

**Pull request merged**